onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -1954,16 +1954,20 @@ def get_prereq(node):
                     if node.output[i_o] in self.sympy_data_:
                         logger.debug('  Sympy Data: ' + str(self.sympy_data_[node.output[i_o]]))
 
-                if (out_shape is not None and None in out_shape) or out_type_undefined:
+                # onnx >= 1.11.0, use unk__#index instead of None when the shape dim is uncertain
+                if (out_shape is not None and None in out_shape) or (out_shape is not None and not all('unk__' not in out for out in out_shape if type(out) == str)) or out_type_undefined:
```

Comment:
Thank you for taking care of it. Is it possible to generalize the case for any symbolic shape instead of only considering 'unk__' prefix?

Although by default ONNX produces symbolic shape with 'unk__' prefix, it can be changed: https://github.com/onnx/onnx/blob/35092895d9bf3592e58f4710d098f8131afef259/onnx/shape_inference/implementation.h#L28 Generalize the case here can help prevent potential issues.

Reply:
I think onnx should create and maintain a specific symbolic shape dim which represent None/uncertain_shape, like "unk__#number". ort need to distinguish whether a string represents none or a variant of the known input dim. If the input dim and the generated symbolic shape all variable, it's difficult to do the distinguish.